### PR TITLE
ci: add action to create doc issue when change labelled

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The change introduces a github action to respond the label: `doc update required`.

When any issue or pull request labelled, it automatically creates an issue in https://github.com/GrepTimeTeam/docs/ to track the doc change. 

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]
LGTM. This action is quite useful ❤️

**Review by waynexia — APPROVED** [review]
I've made an "API Change" lable to do this manually, glad to see it is automated 👍

**Comment by waynexia** [comment]
I'll remove "API Change" and re-tag related to this one.

File: .github/workflows/doc-issue.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Create Issue in docs repo on doc related changes
+
+on:
+  issues:
+    types:
+      - labeled
```
Comment (waynexia) [inline .github/workflows/doc-issue.yml:6]: Trigger this on "pull_request labeled" is also useful 🥲

Comment (sunng87) [reply .github/workflows/doc-issue.yml:6]: yes, in github pull request is an issue too. they shares same sequence number and search system.

Comment (sunng87) [reply .github/workflows/doc-issue.yml:6]: bad, pull request is required explicitly

**Pull request merged**